File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-0.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> calc;
        
        for (int i = 0; i < tokens.size(); i++) {
            if (tokens[i] != "+" && tokens[i] != "-" && tokens[i] != "*" && tokens[i] != "/") {
                int v = stoi(tokens[i]);
                calc.push(v);
            } else {
                int sec = calc.top();
                calc.pop();
                int first = calc.top();
                calc.pop();
                
                if (tokens[i] == "+") {
                    calc.push(first + sec);
                } else if (tokens[i] == "-") {
                    calc.push(first - sec);
                } else if (tokens[i] == "*") {
                    calc.push(first * sec);
                } else if (tokens[i] == "/") {
                    calc.push(first / sec);
                }
            }
        }
        
        return calc.top();
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        if s[0]==')' or s[0]=='}' or s[0]==']':
            return False
        for i in range(len(s)):
            if len(stack)==0:
                stack.append(s[i])
            elif (stack[-1]=='(' and s[i]==')') or (stack[-1]=='{' and s[i]=='}') or (stack[-1]=='[' and s[i]==']') :
                stack.pop()
            else:
                stack.append(s[i])
        if(len(stack)==0):
            return True
        return False

        